pkg/scalers/temporal.go

```diff
@@ -0,0 +1,224 @@
+	}
+
+	metric := GenerateMetricInMili(metricName, float64(queueSize))
+
+	return []external_metrics.ExternalMetricValue{metric}, queueSize > s.metadata.ActivationTargetQueueSize, nil
+}
+
+func (s *temporalScaler) getQueueSize(ctx context.Context) (int64, error) {
+	var selection *sdk.TaskQueueVersionSelection
+	if s.metadata.AllActive || s.metadata.Unversioned || len(s.metadata.BuildIDs) > 0 {
+		selection = &sdk.TaskQueueVersionSelection{
+			AllActive:   s.metadata.AllActive,
+			Unversioned: s.metadata.Unversioned,
+			BuildIDs:    s.metadata.BuildIDs,
+		}
+	}
+
+	queueType := getQueueTypes(s.metadata.QueueTypes)
+
+	resp, err := s.tcl.DescribeTaskQueueEnhanced(ctx, sdk.DescribeTaskQueueEnhancedOptions{
+		TaskQueue:      s.metadata.QueueName,
+		ReportStats:    true,
+		Versions:       selection,
+		TaskQueueTypes: queueType,
+	})
+	if err != nil {
+		return 0, fmt.Errorf("failed to get Temporal queue size: %w", err)
+	}
+
+	return getCombinedBacklogCount(resp), nil
+}
+
+func getQueueTypes(queueTypes []string) []sdk.TaskQueueType {
+	var taskQueueTypes []sdk.TaskQueueType
+	for _, t := range queueTypes {
+		var taskQueueType sdk.TaskQueueType
+		switch t {
+		case "workflow":
+			taskQueueType = sdk.TaskQueueTypeWorkflow
+		case "activity":
+			taskQueueType = sdk.TaskQueueTypeActivity
+		case "nexus":
+			taskQueueType = sdk.TaskQueueTypeNexus
+		}
+		taskQueueTypes = append(taskQueueTypes, taskQueueType)
+	}
+
+	if len(taskQueueTypes) == 0 {
+		return temporalDefauleQueueTypes
+	}
+	return taskQueueTypes
+}
+
+func getCombinedBacklogCount(description sdk.TaskQueueDescription) int64 {
+	var count int64
+	for _, versionInfo := range description.VersionsInfo {
+		for _, typeInfo := range versionInfo.TypesInfo {
+			if typeInfo.Stats != nil {
+				count += typeInfo.Stats.ApproximateBacklogCount
+			}
+		}
+	}
+	return count
+}
+
+func getTemporalClient(ctx context.Context, meta *temporalMetadata) (sdk.Client, error) {
+	options := sdk.Options{
+		HostPort:  meta.Endpoint,
+		Namespace: meta.Namespace,
+		Logger:    sdklog.NewStructuredLogger(slog.Default()),
+	}
+
+	dialOptions := []grpc.DialOption{
+		grpc.WithConnectParams(grpc.ConnectParams{
+			MinConnectTimeout: 5 * time.Second,
+		}),
+	}
+
+	if meta.APIKey != "" {
+		dialOptions = append(dialOptions, grpc.WithUnaryInterceptor(
+			func(ctx context.Context, method string, req any, reply any,
+				cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+				return invoker(
+					metadata.AppendToOutgoingContext(ctx, "temporal-namespace", meta.Namespace),
+					method,
+					req,
+					reply,
+					cc,
+					opts...,
+				)
+			},
+		))
+		options.Credentials = sdk.NewAPIKeyStaticCredentials(meta.APIKey)
+	}
+
+	options.ConnectionOptions = sdk.ConnectionOptions{
+		DialOptions: dialOptions,
+	}
+
+	return sdk.DialContext(ctx, options)
+}
+
+func parseTemporalMetadata(config *scalersconfig.ScalerConfig, _ logr.Logger) (*temporalMetadata, error) {
```

Comment:
This should support parsing the scaled object metadata to infer Temporal configuration from the deployment, rather than requiring all of that to be duplicated into the scaled object.